### PR TITLE
[177] Add missing reconnect tools for composition edges

### PR DESCRIPTION
**Pull request by jerome-obeo**
Bug: https://github.com/eclipse-syson/syson/issues/177

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -24,6 +24,7 @@
 - https://github.com/eclipse-syson/syson/issues/160[#160] [syson] Add explicit usage of node to call syside javascript file.
 - https://github.com/eclipse-syson/syson/issues/172[#172] [details] Add Membership#visibility attribute to Core properties in the Details view.
 - https://github.com/eclipse-syson/syson/issues/171[#171] [details] Add PortUsage#direction attribute to Core properties in the Details view.
+- https://github.com/eclipse-syson/syson/issues/177[#177] [diagrams] Missing reconnect tools for composition edges
```
Comment (AxelRICHARD) [inline CHANGELOG.adoc:27]: Could you the message for something like: `Add reconnect tools for composition edges` please?

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java [hunk 2]
```diff
@@ -98,4 +103,29 @@ private EdgeStyle createEdgeStyle() {
                 .targetArrowStyle(ArrowStyle.NONE)
                 .build();
     }
+
+    private SourceEdgeEndReconnectionTool createSourceReconnectTool() {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java:107]: May be this method could be moved into `AbstractEdgeDescriptionProvider`? What do you think about it?

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java [hunk 3]
```diff
@@ -98,4 +103,29 @@ private EdgeStyle createEdgeStyle() {
...
+        var callService = this.viewBuilderHelper.newChangeContext()
+                .expression(AQLConstants.AQL + AQLConstants.EDGE_SEMANTIC_ELEMENT + ".reconnnectSourceCompositionEdge(" + String.join(",", params) + ")");
+
+        return builder
+                .name("Reconnect Source")
+                .body(callService.build())
+                .build();
+    }
+
+    private TargetEdgeEndReconnectionTool createTargetReconnectTool() {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java:120]: May be this method could be moved into AbstractEdgeDescriptionProvider? What do you think about it?

File: CHANGELOG.adoc (hunk 1 above)
Comment (jerome-obeo) [reply CHANGELOG.adoc:27]: Done. The issue title and the commit message have been changed consequently.

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java (hunk 2 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java:107]: Agree. Done.

File: CHANGELOG.adoc (hunk 1 above)
Comment (AxelRICHARD) [reply CHANGELOG.adoc:27]: But could you also change the message in the changelog please ?

Comment (AxelRICHARD) [reply CHANGELOG.adoc:27]: No you didn't, in the changelog, the message is still `Missing reconnect tools for composition edges`

Comment (jerome-obeo) [reply CHANGELOG.adoc]: I've just pushed my review changes now.

Comment (jerome-obeo) [reply CHANGELOG.adoc]: I marked it as `Done` only to know that I agree and the change is going to be done and pushed with an amended commit.

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**